### ISSUE TITLE
Mutation struct variable empty?

### ISSUE THREAD

**Issue by jorgesece** [shurcooL/graphql#19]
Hi! when I try to execute a mutation with a list of ProductInput structs, I receive in the server a list of products but the attributes (Name and Location) are empty. According to the [documentation](https://github.com/shurcooL/graphql), it is the way to do it, what I'm doing wrong? Many thanks:

```
       type ProductInput struct {
	     Name     graphql.String
	     Location graphql.String
       }
	listProducts := []*ProductInput{}
	for _, element := range installInfo.Package.Products {
		listProducts = append(listProducts, &ProductInput{
			Name:     graphql.String(element.Name),
			Location: graphql.String(element.Image),
		})
	}

	var mutation struct {
		CreateInstallation struct {
			ID string
		} `graphql:"CreateInstallation(installation:{id: $installation, package: {products:$products}})"`
	}
	variables := map[string]interface{}{
		"installation": graphql.UUID(installInfo.UUID.String()),
		"products": listProducts,
	} 
        if err := k.client.Mutation(&mutation, variables); err != nil {
		return err
	}
```

**Comment by shurcooL** [comment]
> I receive in the server

As I understand, you have your own GraphQL server where the client is sending the mutation.

I'm not seeing you do anything clearly wrong in the snippet you posted. It looks like it should work.

Are you sure the data in `installInfo.Package.Products` has non-empty values? Try adding some log statements there to confirm.

The code in `graphql` package that marshals the variables is very simple:

https://github.com/shurcooL/graphql/blob/3d276b9dcc6b1e0adf19557a8de5cb8632c07697/graphql.go#L55-L66

So it should work. See a demo on the playground: https://play.golang.org/p/LIofG3_Gq1g.

What GraphQL server are you using? Can you dump the raw mutation it receives and see if it looks as expected?

**Comment by shurcooL** [comment]
Thanks for following up. I'm glad you got to the bottom of this.

I think I understand what happened now. This general GraphQL client package was factored out of `graphql`, the GitHub-specific GraphQL client, which I created first. Over there, all the input structs were [generated](https://github.com/shurcooL/githubql/blob/d8297a7d5a840a2105de3f27796a6876d03b5da1/input.go#L12-L18), and they include the `json` tags. So it's something I completely forgot about that needs to be done in the general case as well.

This is definitely not intuitive, and currently not documented. I should either document this so it's clear this must be done, or improve the `graphql` code to make it no longer neccessary to manually specify those `json` tags. They can be inferred from the field name, just need to convert from Go's MixedCaps case to GraphQL's lowerCamelCase (this is already being done for queries, the code for it exists in [`ident`](https://godoc.org/github.com/shurcooL/graphql/ident) package).

I filed [issue 20](https://github.com/shurcooL/graphql/issues/20) for that. Thanks again for the report.
